airmail_indexer/src/wof.rs

```diff
@@ -0,0 +1,240 @@
+use anyhow::Result;
+use log::trace;
+use serde::{Deserialize, Serialize};
+use sqlx::{
+    sqlite::{SqliteConnectOptions, SqliteJournalMode, SqlitePoolOptions},
+    Pool, Sqlite,
+};
+use std::path::Path;
+
+/// Performs simple point-in-polygon queries against the WhosOnFirst database.
+/// Queries against the WOF database require the SQLite mod_spatialite extension to be loaded.
+/// Requires package: libsqlite3-mod-spatialite on Debian/Ubuntu
+#[derive(Clone)]
+pub struct WhosOnFirst {
+    pool: Pool<Sqlite>,
+}
+
+impl WhosOnFirst {
+    /// Opens a connection to the WhosOnFirst database.
+    /// Requires the SQLite mod_spatialite extension to be loaded.
+    pub async fn new(path: &Path) -> Result<Self> {
+        trace!("Opening WhosOnFirst database at {:?}", path);
+
+        let opts = SqliteConnectOptions::new()
+            .filename(path)
+            .journal_mode(SqliteJournalMode::Wal)
+            .pragma("cache_size", "2000")
+            .pragma("synchronous", "OFF")
+            .pragma("temp_store", "MEMORY")
+            // .pragma("mmap_size", "1073741824") // 1GB, virtual memory
```

Comment:
nit: consider removing commented lines. in this case I can imagine it being nice to have these here for tweaking so it might make sense to keep them if that's the intent.

Reply:
I'll remove the line. A high mmap value seems to OOM the process faster for planet imports, but it's a little faster for regions.

So... I think in this case, using the defaults for sqlite seems to be a good tradeoff between performance, reasonable memory usage and being reaped by the OS. But this annoys me still (I usually don't care about performance massively...), yhe slowest part of the code seems to be in the sqlite queries. But I'd expect these to be roughly constant time, index backed queries... Something is going on that doesn't quite fit, so I'll likely continue to look into this code hotspot. Possibly too many random pages to the underlying data across the concurrent processes? Not sure tbh